src/components/NewTaxonomy/NewTaxonomy.tsx

```diff
@@ -133,14 +139,57 @@ const NewTaxonomy = ({
     return onLoadData?.(node.value.split(separator));
   }, []);
 
+  const handleSearch = useCallback((list: AntTaxonomyItem[], expandedKeys: React.Key[]) => {
+    setFilteredTreeData(list);
+    setExpandedKeys(expandedKeys);
+  }, []);
+
+  const renderDropdown = useCallback((origin: ReactNode) => {
+    if (!defaultSearch) {
+      return (
+        <TaxonomySearch
+          ref={refInput}
+          treeData={treeData}
+          origin={origin}
```

Comment:
`origin` can be moved out of the TaxonomySearch because it's not used there. will make code cleaner there and simplier here